be/src/exprs/like_predicate.cpp

```diff
@@ -319,10 +388,18 @@ StatusOr<ColumnPtr> LikePredicate::constant_substring_fn(FunctionContext* contex
     if (columns[0]->is_constant()) {
         Slice haystack = ColumnHelper::get_const_value<TYPE_VARCHAR>(columns[0]);
         /// It is assumed that the StringSearcher is not very difficult to initialize.
-        auto searcher = LibcASCIICaseSensitiveStringSearcher(needle.data, needle.size);
+        const char* res_pointer = nullptr;
+        if (state -> ignore_case) {
+            auto searcher = LibcASCIICaseInsensitiveStringSearcher(needle.data, needle.size);
```

Comment:
how about combining the case insensitive function in LibcASCIICaseSensitiveStringSearcher.